core/services/gateway/connector/connector.go

```diff
@@ -27,20 +31,16 @@ type GatewayConnector interface {
 
 // Signer implementation needs to be provided by a GatewayConnector user (node)
 // in order to sign handshake messages with node's private key.
-
-//go:generate mockery --quiet --name Signer --output ./mocks/ --case=underscore
 type Signer interface {
 	// Sign keccak256 hash of data.
 	Sign(data ...[]byte) ([]byte, error)
 }
 
 // GatewayConnector user (node) implements application logic in the Handler interface.
-
-//go:generate mockery --quiet --name GatewayConnectorHandler --output ./mocks/ --case=underscore
 type GatewayConnectorHandler interface {
 	job.ServiceCtx
 
-	HandleGatewayMessage(gatewayId string, msg *api.Message)
+	HandleGatewayMessage(ctx context.Context, gatewayId string, body *api.MessageBody)
```

Comment:
I'm not a huge fan of this. What if there's a product that chooses not to validate signatures but use auth keys instead?
In the Gateway handler we pass full Message objects so let's at least do that here too, for consistency. I'll leave it up to you if you prefer to still Validate() before even calling Handle.

Reply:
> but use auth keys instead?

As I mentioned in another comment above, `api.Message.Signature` is a transport layer signature. It has nothing to do with product-specific logic/needs. That's how I see this. Let's discuss if your vision is different.
I believe that `core/services/gateway/connector/connector.go` should hide all signing and signature validation begin a part of the transport layer. The product-specific handler can do anything else, but should not care about transport needs. We already provide `.Sign()` method that the transport shall use.

Reply:
I wouldn't necessarily consider it a transport-layer signature because the transport doesn't depend on it. The channel between gateway and node is already authenticated on both sides. User authentication is up to the product in my view. The only thing that the transport really needs is the don_id to route messages to correct handler and that's it. I wanted to make everything else flexible.

Maybe some future Gateway use case would not care about signatures but require an auth key in the payload? Maybe some other future case will open the handler up to any user and only rate-limit globally? Who knows.

We can decide to follow your suggestion and treat the signature as a required part of every message and that will be fine for now. I agree that it is a cleaner and more consistent approach but the cost is reduced flexibility.